misk-hibernate-testing/build.gradle

```diff
@@ -6,12 +6,6 @@ dependencies {
   testImplementation dep.openTracingDatadog
 }
 
-test {
-  // To help that with parallel execution by reusing test setup,
-  // misk-hibernate-testing is made dependent on misk-hibernate
-  dependsOn ':misk-hibernate:test'
-}
```

Comment:
add context on why you are doing this?

Reply:
The test dependency on misk-hibernate causes gradle to assume that SpanInjectorTest is inside misk-hibernate when it's actually in misk-hibernate-testing. As a result SpanInjector Test fails to run.

Reply:
Looks like this makes it so that if you attempt to run just tests in `misk-hibernate-testing` it'll trigger tests in `misk-hibernate` to run first. And then this messes up because Intellij added a filtering expression to gradle for a test in `misk-hibernate-testing` for a class that isn't in `misk-hibernate`.

I think this optimization may not even be needed anymore. Seems safe to remove.